### PR TITLE
[OSPK8-423] move osmacaddr CR create from osctlplane to osnetcfg and fix kuttl tests

### PR DESCRIPTION
**Pull request by stuggi**
This is a first step and moves the osmacaddr CR handling from the ctlplane to the osnetcfg CRD/controller.

Also fixes the kuttl tests after the changes we did in the last couple of weeks.

### CODE REVIEW

File: controllers/openstacknetconfig_controller.go [hunk 1]
```diff
@@ -388,16 +403,26 @@ func (r *OpenStackNetConfigReconciler) applyNetAttachmentConfig(
 
 	op, err := controllerutil.CreateOrUpdate(context.Background(), r.Client, attachConfig, apply)
 	if err != nil {
-		err = common.WrapErrorForObject(fmt.Sprintf("Updating %s OpenStackNetworkAttach", attachConfig.Name), attachConfig, err)
+		cond.Message = fmt.Sprintf("Failed to create or update %s %s ", attachConfig.Kind, attachConfig.Name)
+		cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.OpenStackProvisionServerCondReasonCreateError)
```
Comment (abays) [inline controllers/openstacknetconfig_controller.go:407]: Looks like some left-overs here from copy-paste: `OpenStackProvisionServerCondReasonCreateError`

File: controllers/openstacknetconfig_controller.go [hunk 2]
```diff
@@ -529,17 +554,26 @@ func (r *OpenStackNetConfigReconciler) applyNetConfig(
 
 	op, err := controllerutil.CreateOrUpdate(context.Background(), r.Client, osNet, apply)
 	if err != nil {
-		err = common.WrapErrorForObject(fmt.Sprintf("Updating %s OpenStackNetwork", osNet.Name), osNet, err)
+		cond.Message = fmt.Sprintf("Failed to create or update %s %s ", osNet.Kind, osNet.Name)
+		cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.OpenStackProvisionServerCondReasonCreateError)
```
Comment (abays) [inline controllers/openstacknetconfig_controller.go:558]: More copy-paste left-overs? (`OpenStackProvisionServerCondReasonCreateError`)

Comment (stuggi) [reply controllers/openstacknetconfig_controller.go:558]: yes, good catch. will fix it

File: controllers/openstacknetconfig_controller.go (hunk 1 above)
Comment (stuggi) [reply controllers/openstacknetconfig_controller.go]: fixes in latest commit

File: controllers/openstacknetconfig_controller.go (hunk 2 above)
Comment (stuggi) [reply controllers/openstacknetconfig_controller.go]: fixes in latest commit

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**